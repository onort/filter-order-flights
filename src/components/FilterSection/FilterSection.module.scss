@import "../../styles/main.scss";

.container {
  padding: map-get($spacers, normal);
  padding-bottom: map-get($spacers, large);
  border-bottom: 1px solid $muted-border;
}

.header {
  display: flex;
  justify-content: space-between;
  margin-bottom: map-get($spacers, normal);
}

.title {
  display: block;
  color: $grey1;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
  font-weight: 600;
  text-transform: uppercase;
}

.selectAll {
  display: block;
  color: #bdc0c8;
  cursor: pointer;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
  text-decoration: underline;
}
