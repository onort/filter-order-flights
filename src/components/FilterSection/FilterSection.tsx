import React from "react"
import cx from "classnames"

import styles from "./FilterSection.module.scss"

interface Props {
  className?: string
  canSelectAll?: boolean
  infoText?: string
  onSelectAll?: (e: React.MouseEvent<HTMLElement>) => void
  title: string
}

const FilterSection: React.SFC<Props> = props => {
  const { canSelectAll, infoText, onSelectAll, title } = props
  const classNames = cx(styles.container, props.className)
  return (
    <div className={classNames}>
      <div className={styles.header}>
        <span className={styles.title}>{title}</span>
        {infoText && <span className={styles.infoText}>{infoText}</span>}
        {canSelectAll && (
          <span className={styles.selectAll} onClick={onSelectAll}>
            Tümünü Seç
          </span>
        )}
      </div>
      <div className={styles.body}>{props.children}</div>
    </div>
  )
}

FilterSection.defaultProps = {
  canSelectAll: false
}

export default FilterSection
