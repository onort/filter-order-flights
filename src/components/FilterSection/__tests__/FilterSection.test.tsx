import React from "react"
import { shallow } from "enzyme"

import FilterSection from "../"

describe("<FilterSection />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<FilterSection title="Test" />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should display display title", () => {
    const testTitle = "Test Title"
    const wrapper = shallow(
      <FilterSection title={testTitle} canSelectAll={true} />
    )
    expect(wrapper.find(".title").exists()).toBe(true)
    expect(wrapper.find(".title").text()).toBe(testTitle)
  })

  it("should display select all option if prop has been passed", () => {
    const wrapper = shallow(<FilterSection title="Test" canSelectAll={true} />)
    expect(wrapper.find(".selectAll").exists()).toBe(true)
    expect(wrapper.find(".selectAll").text()).toBe("Tümünü Seç")
  })

  it("should fire onSelectAll when select all span is clicked", () => {
    const handleSelectAll = jest.fn()
    const wrapper = shallow(
      <FilterSection
        title="Test"
        canSelectAll={true}
        onSelectAll={handleSelectAll}
      />
    )
    wrapper.find(".selectAll").simulate("click")
    expect(handleSelectAll).toBeCalledTimes(1)
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <FilterSection title="Test" className={testClassName} />
    )
    expect(wrapper.find("div.container").hasClass(testClassName)).toBe(true)
  })
})
