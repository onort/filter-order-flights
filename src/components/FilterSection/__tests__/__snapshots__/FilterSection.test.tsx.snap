// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSection /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterSection
    canSelectAll={false}
    title="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <span
            className="title"
          >
            Test
          </span>
        </div>,
        <div
          className="body"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="title"
            >
              Test
            </span>,
            false,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "title",
            },
            "ref": null,
            "rendered": "Test",
            "type": "span",
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "body",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <span
              className="title"
            >
              Test
            </span>
          </div>,
          <div
            className="body"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="title"
              >
                Test
              </span>,
              false,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "title",
              },
              "ref": null,
              "rendered": "Test",
              "type": "span",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "body",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<FilterSection /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Test</span></div><div class=\\"body\\"></div></div>"`;
